blockly-core/core/ui/contract_editor/x_button.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+goog.provide('Blockly.XButton');
+
+/**
+ * A DOM-based input which sends events on changes
+ * @param {!Object} options
+ *          onNameChanged (newName: String)
+ *          onEnterPressed
+ * @constructor
+ */
+Blockly.XButton = function (options) {
+  this.onButtonPressed = options.onButtonPressed;
+
+  // For disposal
+  this.eventsToUnbind_ = [];
+  this.buttonElement_ = null;
+};
+
+Blockly.XButton.prototype.render = function (parent) {
+  var buttonElement = goog.dom.createDom('button');
+  buttonElement.className = "btn";
+  buttonElement.innerHTML = "x";
+  buttonElement.style.marginRight = "-10px";
+
+  parent.appendChild(buttonElement);
+
+  this.eventsToUnbind_.push(Blockly.bindEvent_(buttonElement, 'mousedown', this, goog.bind(function () {
+    if (this.onButtonPressed) {
+      this.onButtonPressed();
+    }
```

Comment:
I like that style, though we might want a rule of limit-1-null-check for one-liners to prevent really long chains of those (I've seen them become unreadable and difficult to debug when there are multiple null checks leading a function call)